clients/cobol-lsp-vscode-extension/src/commands/RunAnalysisCLI.ts

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+import { Terminal } from "vscode";
+
+export interface AnalysisResults {
+  typeToRun: string;
+  copybookLocation: string;
+}
+
+/**
+ * Class containing functions used to create the command needed to run the analysis from the command line
+ * Handles Java or Native builds with or without copybook support
+ */
+export class RunAnalysis {
+  protected runNative: boolean;
+  protected copybookConfigLocation: string;
+  protected globalStorageUri: vscode.Uri;
+
+  constructor(
+    globalStorageUri: vscode.Uri,
+    runNative: boolean = false,
+    copybookConfigLocation: string = "",
+  ) {
+    this.runNative = runNative;
+    this.copybookConfigLocation = copybookConfigLocation;
+    this.globalStorageUri = globalStorageUri;
```

Comment:
I am not 100% sure if this is the right place for the temp files... nobody is deleting them.
@VitGottwald

Reply:
This is where I was told to write the files to by @VitGottwald after being asked to remove `fs-temp`.

Reply:
I believe this is a good place. The missing piece is a cleanup on startup or shutdown.

Reply:
Added functionality to create a folder within the extension's context to store these files in the latest commit. Also added something such that it will delete any files held within said folder if it does exist, stopping them from piling up. As it is adding the `tempAnalysisFiles` folder name to the provided URI, even if the `context.ExtensionContext` URI from VSCode results in something other than we expect, this deletion will not negatively impact anything.